chain/client/src/client_actor.rs

```diff
@@ -911,6 +917,68 @@ impl ClientActor {
         }
         Ok(blocks_debug)
     }
+
+    /// Returns debugging information about the validator - including things like which approvals were received, which blocks/chunks will be 
+    /// produced and some detailed timing information.
+    fn get_validator_status(
+        &mut self,
+    ) -> Result<ValidatorStatus, near_chain_primitives::Error> {
+        let head = self.client.chain.head()?;
+        let mut production_map: HashMap<BlockHeight, ProductionAtHeight> = HashMap::new();
+
+        if let Some(signer) = &self.client.validator_signer {
+            let validator_id = signer.validator_id().to_string();
+
+            // We want to show some older blocks (up to DEBUG_PRODUCTION_OLD_BLOCKS_TO_SHOW in the past)
+            // and new blocks (up to the current height for which we've sent approval).
+
+            let max_height = std::cmp::min(std::cmp::max(head.height, self.client.doomslug.get_largest_target_height()), DEBUG_MAX_PRODUCTION_BLOCKS_TO_SHOW);
```

Comment:
You can use `Ord::clamp` instaed of min-maxing by hand:

```
            let max_height = head.height.clamp(
                self.client.doomslug.get_largest_target_height(),
                DEBUG_MAX_PRODUCTION_BLOCKS_TO_SHOW,
            );
```

Also, it seems like this is buggy? Dimensions does not match, you are min-maxing heights and heights diffs.